core/src/test/scala/unit/kafka/server/BrokerRegistrationRequestTest.scala

```diff
@@ -145,25 +146,29 @@ class BrokerRegistrationRequestTest {
   }
 
   @ClusterTest(clusterType = Type.KRAFT, brokers = 0, controllers = 1, metadataVersion = MetadataVersion.IBP_3_3_IV3,
-    serverProperties = Array(new ClusterConfigProperty(key = "zookeeper.metadata.migration.enable", value = "true")))
-  def testRegisterZkWithKRaftOldMetadataVersion(clusterInstance: ClusterInstance): Unit = {
+    serverProperties = Array(new ClusterConfigProperty(key = "zookeeper.metadata.migration.enable", value = "false")))
```

Comment:
Do we have a test of trying to start the controller with `zookeeper.metadata.migration.enable = true` and a bootstrap MV that's too old? It should exit the controller, right? (Due to throwing that RuntimeException)?

It would be good to have an integration test of that...

Reply:
I added a test in QuorumControllerTest -- it's a little hard to integration test since we don't expose the fault handler outside of TestKit (which maybe we should)

Reply:
ok